// @svelteuidev/core/dist/components/AppShell/VerticalSection/VerticalSection.styles.js
import { createStyles } from '../../../styles';
export default createStyles((theme, { height, borderPosition, fixed, position, zIndex }) => {
    return {
        root: {
            [`${theme.dark} &`]: {
                backgroundColor: theme.fn.themeColor('dark', 7),
                borderBottom: borderPosition === 'bottom' ? `1px solid ${theme.fn.themeColor('dark', 5)}` : undefined,
                borderTop: borderPosition === 'top' ? `1px solid ${theme.fn.themeColor('dark', 5)}` : undefined
            },
            ...position,
            zIndex,
            height,
            fontFamily: '$standard' || 'sans-serif',
            maxHeight: height,
            position: fixed ? 'fixed' : 'static',
            boxSizing: 'border-box',
            backgroundColor: 'white',
            borderBottom: borderPosition === 'bottom' ? `1px solid ${theme.fn.themeColor('gray', 2)}` : undefined,
            borderTop: borderPosition === 'top' ? `1px solid ${theme.fn.themeColor('gray', 2)}` : undefined
        }
    };
});
